packages/alchemy/src/react/hooks/useAddPasskey.ts

```diff
@@ -0,0 +1,45 @@
+"use client";
+
+import { useMutation, type UseMutateFunction } from "@tanstack/react-query";
+import { ClientOnlyPropertyError } from "../../config/errors.js";
+import { useAlchemyAccountContext } from "../context.js";
+import { useSigner } from "./useSigner.js";
+
+export type UseAddPasskeyResult = {
+  addPasskey: UseMutateFunction<
+    string[],
+    Error,
+    CredentialCreationOptions | undefined | void,
+    unknown
+  >;
+  isAddingPasskey: boolean;
+  error: Error | null;
+};
+
+export function useAddPasskey(): UseAddPasskeyResult {
+  const { queryClient } = useAlchemyAccountContext();
+  const signer = useSigner();
+
+  const {
+    mutate: addPasskey,
+    isPending: isAddingPasskey,
+    error,
+  } = useMutation(
+    {
+      mutationFn: async (params?: CredentialCreationOptions | void) => {
+        if (!signer) {
+          throw new ClientOnlyPropertyError("signer");
+        }
```

Comment:
nit: you shouldn't need this because `useSigner` already handles that for you

Reply:
as a result you should be able to update this to just be:

```
useMutation({
  mutationFn: signer.addPasskey
})
```

Reply:
it's safe to assume that `useMutation` will run on the client only. so you can probs get away with doing `signer!.`

Reply:
oh i see, ok yea that makes sense